src/lib.rs

```diff
@@ -639,6 +650,75 @@ impl CertificateSigningRequest {
 	}
 }
 
+/// A certificate revocation list (CRL)
+///
+/// ## Example
+///
+/// ```
+/// extern crate rcgen;
+/// use rcgen::*;
+///
+/// # fn main () {
+/// // Generate a CRL issuer.
+/// let mut issuer_params = CertificateParams::new(vec!["crl.issuer.example.com".to_string()]);
+/// issuer_params.is_ca = IsCa::Ca(BasicConstraints::Unconstrained);
+/// issuer_params.key_usages = vec![KeyUsagePurpose::KeyCertSign, KeyUsagePurpose::DigitalSignature, KeyUsagePurpose::CrlSign];
+/// let issuer = Certificate::from_params(issuer_params).unwrap();
+/// // Describe a revoked certificate.
+/// let revoked_cert = RevokedCertParams{
+///   serial_number: SerialNumber::from(9999),
+///   revocation_time: date_time_ymd(2024, 06, 17),
+///   reason_code: Some(RevocationReason::KeyCompromise),
+///   invalidity_date: None,
+/// };
+/// // Create a CRL signed by the issuer, revoking revoked_cert.
+/// let crl = CertificateRevocationListParams{
+///   this_update: date_time_ymd(2023, 06, 17),
+///   next_update: date_time_ymd(2024, 06, 17),
+///   crl_number: CrlNumber::from(1234),
+///   revoked_certs: vec![revoked_cert],
+///   alg: &PKCS_ECDSA_P256_SHA256,
+///   key_identifier_method: KeyIdMethod::Sha256,
+/// };
+/// let crl = CertificateRevocationList::from_params(crl).unwrap();
+/// println!("{}", crl.serialize_pem_with_signer(&issuer).unwrap());
+///# }
+pub struct CertificateRevocationList {
+	params : CertificateRevocationListParams,
```

Comment:
Style nit: you use `params :CertificateRevocationListParams` here? My idea is that I always do `<space>:<space>` if it's used like an assignment, and `<space>:` if it's indicating a type.

Reply:
Thanks, that one slipped through. Fixed.